openbas-model/src/main/java/io/openbas/config/EngineConfig.java

```diff
@@ -0,0 +1,40 @@
+package io.openbas.config;
+
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConfigurationProperties(prefix = "engine")
+@Data
+public class EngineConfig {
+
+  private String engineSelector = "elk";
+
+  private String indexPrefix = "openbas";
+
+  private String indexSuffix = "-000001";
+
+  private String numberOfShards = "1";
+
+  private String numberOfReplicas = "1";
+
+  private int maxResultWindow = 100000;
+
+  private int defaultPagination = 500;
+
+  private long maxPrimaryShardDocs = 75000000;
+
+  private String maxPrimaryShardsSize = "50Gb";
+
+  private String maxFieldsSize = "4096";
+
+  @NotNull private String url;
+
+  private String username;
+
+  private String password;
+
+  private boolean rejectUnauthorized = false;
```

Comment:
Shouldn't this be `true` by default ?

Reply:
We allow Elasticsearch to run without any authentication configured.
Could this be an issue?

Reply:
If I understand what this flag disables [here](https://github.com/OpenBAS-Platform/openbas/pull/2701/files#diff-41edaaa6cc2ed01b50c34156fb27c37c59f5c20018722f5865d35c81e7e2a288R77), TLS certificate signatures and hostname won't be verified by default.

Most software enable such verification by default, won't users expect that here too ?

Reply:
I made the changes